### PR TITLE
JS DOM task

### PR DESCRIPTION
**Pull request by wwwowka**
### JavaScript DOM

[Demo](https://wwwowka.github.io/task_js_dom/) |
[Code base](https://github.com/wwwowka/task_js_dom)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/wwwowka/task_js_dom/script.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+const menu = document.querySelector(".menu");
+const mainInner = document.querySelector(".content");
+
+function createContent({ name, image, article }) {
+    return (
+        `<div class="content__title"><h2>${name}</h2></div>
+        <div class="content__foto1"><img src="${image[0]}" alt="Фото ${name} 1"></div>
+        <div class="content__foto2"><img src="${image[1]}" alt="Фото ${name} 2"></div>
+        <div class="content__foto3"><img src="${image[2]}" alt="Фото ${name} 3"></div>
+        <div class="content__foto4"><img src="${image[3]}" alt="Фото ${name} 4"></div>
```
Comment (madmaxWMFU) [inline submissions/wwwowka/task_js_dom/script.js:63]: Use a method of array `map` or `reduce`.

File: submissions/wwwowka/task_js_dom/script.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+const changeContent = function (event) {
+    const selectPlant = plant.find((obj) => obj.id === event.target.id);
+    mainInner.innerHTML = createContent(selectPlant);
+};
+
+menu.addEventListener("click", changeContent);
+
+mainInner.innerHTML = createContent(plant[1]);
```
Comment (madmaxWMFU) [inline submissions/wwwowka/task_js_dom/script.js:75]: Create an initial variable for `plant[1]`.
Or try to use destruction and get zero elements of the data list.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Good work. See some comments below.

**Comment by wwwowka** [comment]
Thanks for the review and advice. Made changes.
Didn't understand "Or try to destruction use and get zero elements of the data list."

File: submissions/wwwowka/task_js_dom/script.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        return `<div class="content__foto${i++}"><img src="${img}" alt="Фото ${name} ${i}"></div>`;
+     })
+    .join('');
+
+
+    return (
+        `<div class="content__title"><h2>${name}</h2></div>
+        ${arrayImg}
+       <p class="content__text">${article}</p>`
+    );
```
Comment (madmaxWMFU) [inline submissions/wwwowka/task_js_dom/script.js:71]:
```suggestion
    const arrayImg = image
    .map((img, imgkey)=>
     {
        return `<div class="content__foto${++imgkey}"><img src="${img}" alt="Фото ${name} ${imgkey}"></div>`;
     })
    .join('');

    return (
        `<div class="content__title"><h2>${name}</h2></div>
        ${arrayImg}
       <p class="content__text">${article}</p>`
    );
```

What happen will be array an empty?

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

File: submissions/wwwowka/task_js_dom/script.js (hunk 3 above)
Comment (wwwowka) [reply submissions/wwwowka/task_js_dom/script.js]: Thanks, I've applied the changes.
If the array is empty JS will give an error, DIV will be empty.

Do I need to assign default values to "defaultContent"? Then I need the default images as well.
Or write default values in the function?

Comment (madmaxWMFU) [reply submissions/wwwowka/task_js_dom/script.js]: Error is always bad)
In the future, you must be doing checks for situations like this.
If I correctly understand a question, the `defaultContent` is the first element of your data.

Comment (wwwowka) [reply submissions/wwwowka/task_js_dom/script.js]: Yes, the function is called with a `defaultContent` until the user has made a selection.

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**